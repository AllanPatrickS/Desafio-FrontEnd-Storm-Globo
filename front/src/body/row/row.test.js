import { cleanup } from '@testing-library/react';
import { create } from "react-test-renderer";
import Row from './row';

describe('Testing row', () => {

  it('renders correctly', () => {
    const wrapper = create(<Row row={{
      _id: 'teste',
      username: 'test',
      email: 'email@teste.com',
      createdAt: '20/20/2020',
      updatedAt: '20/20/2020',
      rules: 0,
      status: true
    }}/>)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  afterEach(function () {
    cleanup();
  });

});