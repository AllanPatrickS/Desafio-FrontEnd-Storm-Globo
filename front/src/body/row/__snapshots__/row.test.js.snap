// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing row renders correctly 1`] = `
<tr
  className="MuiTableRow-root makeStyles-oddRow-2"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-10 Mui-checked MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          aria-label="select all desserts"
          checked={true}
          className="PrivateSwitchBase-input-12"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
    </span>
  </td>
  <th
    aria-sort={null}
    className="MuiTableCell-root makeStyles-cell-3"
    role="cell"
    scope="row"
  >
    test
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root makeStyles-cell-3"
  >
    email@teste.com
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root makeStyles-cell-3"
  >
    20/20/2020
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root makeStyles-cell-3"
  >
    20/20/2020
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root makeStyles-cell-3"
  >
    0
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root makeStyles-green-4"
  >
    Ativo
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    <div
      className="MuiBox-root MuiBox-root-16"
    >
      <button
        aria-controls={null}
        aria-haspopup={true}
        aria-label="more"
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-gray-15"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </td>
</tr>
`;
